nxdrive/manager.py

```diff
@@ -558,6 +559,35 @@ def generate_report(self, *, path: Path = None) -> Path:
         report.generate()
         return report.get_path()
 
+    def generate_csv(self, session_id: int, engine: Engine) -> bool:
+        """
+        Generate a CSV file based on the *session_id* in an async Runner.
+        """
+        log.info(f"Generating CSV file from Direct Transfer session {session_id}.")
+
+        session = engine.dao.get_session(session_id)
+        if not session:
+            return False
+
+        runner = Runner(self._generate_csv_async, engine, session)
+        engine._threadpool.start(runner)
+        return True
+
+    def _generate_csv_async(self, engine: Engine, session: Session) -> None:
+        from .session_csv import SessionCsv
+
+        try:
+            session_items = engine.dao.get_session_items(session.uid)
+            session_csv = SessionCsv(self, engine, session)
+            session_csv.create_tmp()
+            engine.dao.sessionUpdated.emit()
+            session_csv.store_data(session_items)
+        except Exception:
+            log.exception("Asynchronous CSV generation error")
+            session_csv.output_file.unlink()
```

Comment:
(in case the exception happens before the file is created)
```suggestion
            session_csv.output_file.unlink(missing_ok=True)
```